### ISSUE TITLE
feature request: ignore colon

### ISSUE THREAD

**Issue by alextes** [launchdarkly/rust-eventsource-client#37]
The server I'm listen to is sending a `:` every now and then. Reading [MDN: Event Stream Format](https://developer.mozilla.org/en-US/docs/Web/API/Server-sent_events/Using_server-sent_events#event_stream_format) it seems this is a way to keep the connection alive. To quote the relevant section directly:
>The event stream is a simple stream of text data which must be encoded using [UTF-8](https://developer.mozilla.org/en-US/docs/Glossary/UTF-8). Messages in the event stream are separated by a pair of newline characters. A colon as the first character of a line is in essence a comment, and is ignored.

>Note: The comment line can be used to prevent connections from timing out; a server can send a comment periodically to keep the connection alive.

Currently, these messages result in this lib returning `InvalidEvent`. Does that sound like something this library should support?

Thanks for the lib!

**Comment by alextes** [comment]
I just noticed in the example there is a `SSE::Comment` maybe I misunderstood something but the reason it appears to me this library is not handling comments is that making a curl request directly its clear the `InvalidEvent` is a line with a single `:`.

Took a quick look at the code, the library is definitely attempting to handle comments. Not sure what's going on, output from curl looks like
```
:

:

event:finalized_checkpoint
data:{"block":"0x40a6aafd38f301bf321858cf002297fc5c68354a0f656f8f86778393af6c5fd4","state":"0x7494da40a5decfb8df096695c5c6fac67c1daf4361708449eada2b79e12b9a86","epoch":"131671"}

:

:

:

:

```

**Comment by keelerm84** [comment]
@alextes I have opened a [PR](https://github.com/launchdarkly/rust-eventsource-client/pull/38) that I believe will address the issue you are experiencing. Would it be possible for you to confirm by pulling from that branch?

**Comment by alextes** [comment]
@keelerm84 that fixes it 👍

**Comment by keelerm84** [comment]
I'm sorry it took us so long to release a fix for this. Resolved in 0.10.2.
